### ISSUE TITLE
Video randomly not loading (black screen)

### ISSUE THREAD

**Issue by multimedia-enpjj** [aframevr/aframe#5407]
I've made a 3D VR experience using A-Frame and oscarmarinmiro's aframe-stereo-component.
There is no issue when using low resolution videos: 1920x1920 videos but, when using 3840x3840 videos, depending on page refreshes, some videos don't load or play when clicking on them, while others do (so it doesn't appear to be an encoding problem - all videos are shot, processed and encoded in the same way).

I have implemented a "log popup", so I can check on errors, but none come up automatically. When I check if the video is playing, the videos that are not working return a redyState of 1 but they do load in buffer (about by the same amount as working videos).

**The problem only occurs in the Oculus Quest 2 (not on PC).**

NB: This web app is online (see link bellow) but is made to run on an offline server (hosted on a raspberry pi).

- A-Frame Version: 1.3.0
- Platform / Device: Oculus Quest 2
- Reproducible Code Snippet or URL: https://multimedia.enpjj.fr/vr/v2/

**Comment by dmarcos** [comment]
Any errors on console when it fails? Can you play the videos consistently on a normal video tag in 2D?

**Comment by dmarcos** [comment]
Also use A-Frame 1.5.0. You're in an old version

**Comment by dmarcos** [comment]
also if resolution is the only difference. You might be hitting some browser / memory limitations. Worth also testing on Quest3. I suspect this might not have anything to do with A-Frame / THREE but not 100% yet.

**Comment by multimedia-enpjj** [comment]
Thanks dmarcos for trying to help!

> Any errors on console when it fails? Can you play the videos consistently on a normal video tag in 2D?

I quickly made this page: https://multimedia.enpjj.fr/vr/v2/videos.html, to check your second suggestion. No problem playing the videos there. As for console errors, there is no way to access the consol in Meta Browser, so I don't know... I've tried using FireFox Reality, because i's supposed to have developer tools, but nothing works properly in it and, despite activating the developer features, I can't find a way to access them >_<

> Also use A-Frame 1.5.0. You're in an old version

I'll try that but I was using a old version because I had to modify it to make everything accessible locally (for it to run on the pi web server - see original post). However, I did (temporarily) switch to version 1.5.0 but, although it messes up some stuff (lighting, and custom ray orientation) and it appears to be incompatible with rozek's aframe-rounded-box-component that I'm also using, it didn't change anything regarding the original video problem...

> also if resolution is the only difference. You might be hitting some browser / memory limitations. Worth also testing on Quest3. I suspect this might not have anything to do with A-Frame / THREE (perhaps aframe-stereo-component implementation but not familiar with it) but not 100% yet.

Luckily, I just bought the Quest 3 but I couldn't really test it because the version of aframe I was using doesn't have Quest 3 controllers implemented... I'll have to try it tonight, at home, with version 1.5.0

**Comment by dmarcos** [comment]
You can debug and see the console on Quest. Can remote debug with your desktop Chrome:

https://developer.oculus.com/documentation/web/browser-remote-debugging/

**Comment by multimedia-enpjj** [comment]
> You can debug and see the console on Quest. Can remote debug with your desktop Chrome:
> https://developer.oculus.com/documentation/web/browser-remote-debugging/

Thanks so much @dmarcos ! Accessing the console allowed me to see that I had two types messages: the first was a simple warning `RENDER WARNING: texture bound to texture unit 0 is not renderable. It might be non-power-of-2 or have incompatible texture filtering (maybe)?` linked, as I discovered, to an actual error `Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause().`

Basically, play() was triggered to early. The quest 2 had not yet "loaded" the videos as textures but I was already trying to play them (even on the local server). That's why lower resolution videos didn't have any problems (they took less time to load).

So, to time everything and make sure every steps take place after the previous one has finished, I implemented the solution suggested here: [https://developer.chrome.com/blog/play-request-was-interrupted](https://developer.chrome.com/blog/play-request-was-interrupted)

I was trying to solve a problem I could not see and, thanks to you I can now see it.
Thanks again!
